framework/wazuh/cluster/dapi/dapi.py

```diff
@@ -0,0 +1,286 @@
+    """
+    Merge results from an API call.
+
+    To do the merging process, the following is considered:
+        1.- If the field is a list, append items to it
+        2.- If the field is a message (msg), only replace it if the new message has more priority.
+
+    The priorities are defined in a list of tuples. The first item of the tuple is the element which has more priority.
+
+    :param responses: list of results from each node
+    :param final_json: JSON to return.
+    :return: single JSON with the final result
+    """
+    priorities = {
+        ("Some agents were not restarted", "All selected agents were restarted")
+    }
+
+    for local_json in responses:
+        for key,field in local_json.items():
+            field_type = type(field)
+            if field_type == dict:
+                final_json[key] = merge_results([field], {} if key not in final_json else final_json[key], input_json)
+            elif field_type == list:
+                if key in final_json:
+                    final_json[key].extend([elem for elem in field if elem not in final_json[key]])
+                else:
+                    final_json[key] = field
+            elif field_type == int:
+                if key in final_json:
+                    if key == 'totalItems':
+                        final_json[key] += field
+                    elif key == 'error' and final_json[key] < field:
+                        final_json[key] = field
+                else:
+                    final_json[key] = field
+            else: # str
+                if key in final_json:
+                    if (field, final_json[key]) in priorities:
+                        final_json[key] = field
+                else:
+                    final_json[key] = field
+
+    if 'data' in final_json and 'items' in final_json['data'] and isinstance(final_json['data']['items'],list):
+        offset,limit = input_json['arguments']['offset'], input_json['arguments']['limit']
+        final_json['data']['items'] = final_json['data']['items'][offset:offset+limit]
+
+    return final_json
+
+
+class APIRequestQueue(communication.ClusterThread):
+    def __init__(self, server, stopper):
+        communication.ClusterThread.__init__(self, stopper=stopper)
+        self.server = server
+        self.request_queue = Queue()
+        self.tag = "[DistributedAPI]"
+
+
+    def run(self):
+        while not self.stopper.is_set() and self.running:
+            name, id, request = self.request_queue.get(block=True).split(' ', 2)
+            result = distribute_function(json.loads(request), from_master=True)
+            try:
+                self.send_string(result, id, name)
+            except Exception as e:
+                self.send_request(command='err-is', data=str(e), id=id, name=name)
+
+
+    def send_string(self, result, id, name):
+        if name == 'None':
+            self.server.send_string(reason='dapi_res', string_data=result, new_req="fwd_new", upd_req="fwd_upd",

```

Comment:
When a `dapi` request is received in a worker node, the name is set to "none" because it is only able to communicate with the master:

https://github.com/wazuh/wazuh/blob/d808f6e7a3ed18976b64988e07228b0344177610/framework/wazuh/cluster/worker.py#L74-L75

However, when received in a master node, the worker name it came from is necessary to know where to send the response:

https://github.com/wazuh/wazuh/blob/d808f6e7a3ed18976b64988e07228b0344177610/framework/wazuh/cluster/master.py#L88-L90

In addition, a worker's `send_string` function doesn't support `name` parameter:
https://github.com/wazuh/wazuh/blob/d808f6e7a3ed18976b64988e07228b0344177610/framework/wazuh/cluster/communication.py#L301

Whereas the master does:
https://github.com/wazuh/wazuh/blob/d808f6e7a3ed18976b64988e07228b0344177610/framework/wazuh/cluster/communication.py#L684